File: 1394. Find Lucky Integer in an Array.py
class Solution:
    def findLucky(self, arr: List[int]) -> int:
        count = [0]*len(arr)
        
        for i in range(len(arr)):
            cur, count[i] = arr[i]-1, count[i]+i+1
            
            if cur<len(count):
                count[cur] -= 1
        
        for i in range(len(count)-1, -1, -1):
            if count[i]==0:
                return i+1
        
        return -1

File: 1394. Find Lucky Integer in an Array.js
/**
 * @param {number[]} arr
 * @return {number}
 */
 var findLucky = function(arr) {
    let count = new Array(arr.length).fill(0);

    for(let i=0;i<arr.length;i++){
        let cur = arr[i]-1;
        count[i] += i+1;

        if(cur<count.length){
            count[cur] -= 1;
        }
    }

    for(let i=count.length-1;i>-1;i--){
        if(count[i]==0){
            return i+1;
        }
    }
    return -1;
};